### PR TITLE
feature(script3) merge with master 

### PR DESCRIPTION
**Pull request by dropsrobs**

### CODE REVIEW

**Comment by spidzio** [comment]
Great, that you cleaned it up. 
Now I can make my comments.

File: script3.rb [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+files_ext = {}
+
+answear = search_in_hidden?
+size_diff = insert_size_diff
+
+files_name = scan_files_names answear, files_name
+files_ext = scan_files_ext answear, files_ext
+
+scan_dup_by_name files_name
+scan_dup_by_size files_ext, size_diff
```
Comment (spidzio) [inline script3.rb]: This solution is way, way, way too much complicated.
This is the Ruby-style solution of this problem:
```ruby
dirname = ARGV[0] || Dir.pwd
size_tsh = ARGV[1].to_i || 10

exit unless File.exist?(dirname) && File.directory?(dirname)

files = []

Dir.glob("#{dirname}/**/*") do |path|
  next unless File.file?(path)
  files << {
    name: File.basename(path),
    path: path,
    size: File.size(path),
    ext: File.extname(path)
  }
end

files.combination(2).each do |pair|
  file1, file2 = pair
  next unless file1[:ext] == file2[:ext]
  size_diff = (file1[:size] - file2[:size]).abs
  next unless file1[:name] == file2[:name] || size_diff < size_tsh
  puts "#{file1[:path]} <-----> #{file2[:path]}"
end
```

1. It's better to assume some default config, and use command line arguments for not default value, when you make a console script
2. If something begins to be complicated in Ruby, it's probably you don't know the simple solution.

**Comment by spidzio** [comment]
Next time please provide some pull request description, and more descriptive title :)

**Comment by dropsrobs** [comment]
I just wanted to see if it works at all :) Next time I'll do this!

File: script3.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require 'find'
+
+files = []
+dirname = ARGV[0] || Dir.pwd
+usr_size_diff = ARGV[1] || 10
+usr_size_diff = usr_size_diff.to_i
```
Comment (spidzio) [inline script3.rb]: Good, that you've noticed it: `to_i`

File: script3.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+require 'find'
```
Comment (spidzio) [inline script3.rb]: What is this `require` for?

**Comment by spidzio** [comment]
Please write a comment with short description of what have you changed, when you push something to the pull request.

**Comment by dropsrobs** [comment]
Removed unecessary *require 'find'*.

File: testDir/test_file.rb [hunk 4]
```diff
@@ -0,0 +1 @@
+
```
Comment (spidzio) [inline testDir/test_file.rb]: There are Rubocop offences in the test files. 
It is great opportunity to change default rubocop configuration and exclude test directories. 
It is quite simle, so pleas do it.

**Comment by dropsrobs** [comment]
Added .rubocop.yml configuration file.

File: .rubocop.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+AllCops:
+  Exclude:
+    - 'testDir/**/*'
```
Comment (spidzio) [inline .rubocop.yml]: Nice

**Comment by spidzio** [comment]
:+1:

**Pull request merged**